microservices/site-launch/lambda-functions/redirection-domain-validation/index.ts

```diff
@@ -55,34 +59,35 @@ export const redirectionDomainValidation = async (
         ref: DEFAULT_BRANCH,
       }
     )
-    if (response.status === 200) return // file already exist
-
-    if (response.status !== 404) {
-      const error = `Unexpected error occurred for redirection for ${primaryDomainSource} : 
-      ${JSON.stringify(response)}`
-      throw new Error(error)
+  } catch (error) {
+    if (
+      Object.prototype.hasOwnProperty.call(error, "status") &&
+      error.status === 404
+    ) {
+      fileExists = false
+    } else {
+      throw Error(
+        `Unknown error when checking for file existence of ${primaryDomainSource}.conf: ${error}`
+      )
     }
-
-    response = await octokit.request(
-      `PUT /repos/isomerpages/isomer-redirection/contents/letsencrypt/${primaryDomainSource}.conf`,
-      {
-        owner: "isomerpages",
-        repo: "isomer-redirection",
-        path: "letsencrypt/.",
-        message: `Create ${primaryDomainSource}.conf`,
-        committer: {
-          name: "isomeradmin",
-          email: "isomeradmin@open.gov.sg",
-        },
-        content: Buffer.from(template, "binary").toString("base64"),
-        branch: DEFAULT_BRANCH,
-      }
-    )
-    logger.info(
-      `status of redirection commit for ${primaryDomainSource}:\n ${response}`
-    )
-  } catch (err) {
-    logger.error(err)
-    throw err
   }
+  if (fileExists) return
+  const response = await octokit.request(
+    `PUT /repos/isomerpages/isomer-redirection/contents/letsencrypt/${primaryDomainSource}.conf`,
+    {
+      owner: "isomerpages",
+      repo: "isomer-redirection",
+      path: "letsencrypt/.",
+      message: `Create ${primaryDomainSource}.conf`,
+      committer: {
+        name: "isomeradmin",
+        email: "isomeradmin@open.gov.sg",
+      },
+      content: Buffer.from(template, "binary").toString("base64"),
+      branch: DEFAULT_BRANCH,
+    }
+  )
+  logger.info(
+    `status of redirection commit for ${primaryDomainSource}:\n ${response}`
+  )
```

Comment:
8b6c8e81
you are right, this is meaningless bloat since this line will actually only be executed if the previous `await octokit.request` is a success! (any errors will result cause the error to be thrown, which the step functions will catch)